### PR TITLE
Address Review of v0.12

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Review Cleanup

**What does this PR do? Why is it needed?**

Fixes some issues from review of #5614 

- [x] Refactor broadcaster to use the same method for message broadcasting.
- [x]  Properly annotate errors for attester rpc methods
- [x] Clean up comments.

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

N.A

### CODE REVIEW

File: beacon-chain/rpc/validator/attester.go [hunk 1]
```diff
@@ -211,19 +217,22 @@ func (vs *Server) SubscribeCommitteeSubnets(ctx context.Context, req *ethpb.Comm
 		return uint64(len(vals)), nil
 	}
 
-	// run first request
+	// request the head validator indices of
```
Comment (terencechain) [inline beacon-chain/rpc/validator/attester.go:220]: Why are the comments formatted like this?

Comment (nisdas) [reply beacon-chain/rpc/validator/attester.go]: ok changed

File: beacon-chain/rpc/validator/attester.go [hunk 2]
```diff
@@ -211,19 +217,21 @@ func (vs *Server) SubscribeCommitteeSubnets(ctx context.Context, req *ethpb.Comm
 		return uint64(len(vals)), nil
 	}
 
-	// run first request
+	// Request the head validator indices of epoch represented by the first requested
+	// slot.
 	currValsLen, err := fetchValsLen(req.Slots[0])
 	if err != nil {
-		return nil, err
+		return nil, status.Errorf(codes.Internal, "Could not retrieve head validator indices: %v", err)
```
Comment (terencechain) [inline beacon-chain/rpc/validator/attester.go:224]:
```suggestion
		return nil, status.Errorf(codes.Internal, "Could not retrieve head validator length: %v", err)
```

`fetchValsLen` fetches the head validator length

File: beacon-chain/rpc/validator/attester.go [hunk 3]
```diff
@@ -211,19 +217,21 @@ func (vs *Server) SubscribeCommitteeSubnets(ctx context.Context, req *ethpb.Comm
...
 	}
 	currEpoch := helpers.SlotToEpoch(req.Slots[0])
 
 	for i := 0; i < len(req.Slots); i++ {
-		// if epoch has changed, re-request active validators
+		// If epoch has changed, re-request active validators
```
Comment (terencechain) [inline beacon-chain/rpc/validator/attester.go:229]:
```suggestion
		// If epoch has changed, re-request active validators length
```

File: beacon-chain/rpc/validator/attester.go [hunk 4]
```diff
@@ -211,19 +217,21 @@ func (vs *Server) SubscribeCommitteeSubnets(ctx context.Context, req *ethpb.Comm
...
+		// indices.
```
Comment (terencechain) [inline beacon-chain/rpc/validator/attester.go:230]:
```suggestion
```

File: beacon-chain/rpc/validator/attester.go [hunk 5]
```diff
@@ -211,19 +217,21 @@ func (vs *Server) SubscribeCommitteeSubnets(ctx context.Context, req *ethpb.Comm
...
 		if currEpoch != helpers.SlotToEpoch(req.Slots[i]) {
 			currValsLen, err = fetchValsLen(req.Slots[i])
 			if err != nil {
-				return nil, err
+				return nil, status.Errorf(codes.Internal, "Could not retrieve head validator indices: %v", err)
```
Comment (terencechain) [inline beacon-chain/rpc/validator/attester.go:234]:
```suggestion
				return nil, status.Errorf(codes.Internal, "Could not retrieve head validator length: %v", err)
```

File: beacon-chain/rpc/validator/attester.go [hunk 6]
```diff
@@ -195,11 +198,14 @@ func (vs *Server) ProposeAttestation(ctx context.Context, att *ethpb.Attestation
 
 // SubscribeCommitteeSubnets subscribes to the committee ID subnet given subscribe request.
 func (vs *Server) SubscribeCommitteeSubnets(ctx context.Context, req *ethpb.CommitteeSubnetsSubscribeRequest) (*ptypes.Empty, error) {
+	ctx, span := trace.StartSpan(ctx, "AttesterServer.SubscribeCommitteeSubnets")
+	defer span.End()
+
 	if len(req.Slots) != len(req.CommitteeIds) && len(req.CommitteeIds) != len(req.IsAggregator) {
 		return nil, status.Error(codes.InvalidArgument, "request fields are not the same length")
 	}
 	if len(req.Slots) == 0 {
```
Comment (terencechain) [inline beacon-chain/rpc/validator/attester.go:207]: what happen during genesis when attesters request for slot 0?

Comment (rauljordan) [reply beacon-chain/rpc/validator/attester.go:207]: req.Slots will still have len 1 in this case

Comment (terencechain) [reply beacon-chain/rpc/validator/attester.go:207]: ah yes, i was thinking the actual slot number. my b!

**Pull request merged**